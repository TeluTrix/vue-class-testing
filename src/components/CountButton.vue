<script setup lang="ts">
import { ref } from "vue";

const count = ref(0);
</script>

<template>
  <div>
    <button id="increment" type="button" class="actionBtn" @click="count++">Minus(-)</button>
    <p>count is: {{ count }}</p>
    <button id="decrement" type="button" class="actionBtn" @click="count--">Plus(+)</button>
  </div>
</template>

<style scoped>
.actionBtn {
  margin: 5pt;
  padding: 5pt;
}
</style>
