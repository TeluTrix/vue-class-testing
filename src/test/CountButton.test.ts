/* import { expect, test } from "vitest";
import CountButton from "../components/CountButton.vue";
import { mount, VueWrapper } from "@vue/test-utils";
import { ComponentPublicInstance } from "vue"; */

// eslint-disable-next-line @typescript-eslint/no-var-requires
const xTimes = require("js-utility-x");
/*
test("Mounted Components", async () => {
  const wrapper: VueWrapper<ComponentPublicInstance> = mount(CountButton);
  expect(CountButton).toBeTruthy();

  //await wrapper.get("#decrement").trigger("click");
  //expect(wrapper.text()).toContain("0");
  //expect(wrapper.html()).toMatchSnapshot();
  //await wrapper.get("#increment").trigger("click");
  await xTimes(() => wrapper.get("#increment").trigger("click"), 9);
  //await wrapper.get("#increment").trigger("click");
  expect(wrapper.text()).toContain("9");

  //await wrapper.get("#increment").trigger("click");
});

test("Minus Component", async () => {
  const wrapper: VueWrapper<ComponentPublicInstance> = mount(CountButton);
  expect(CountButton).toBeTruthy();
  await wrapper.get("#decrement").trigger("click");
  expect(wrapper.text()).toContain("-1");
});

test("Plus Component", async () => {
  const wrapper: VueWrapper<ComponentPublicInstance> = mount(CountButton);
  expect(CountButton).toBeTruthy();
  await wrapper.get("#increment").trigger("click");
  expect(wrapper.text()).toContain("1");
});
*/
