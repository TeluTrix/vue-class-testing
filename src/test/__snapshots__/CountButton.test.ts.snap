// Vitest Snapshot v1

exports[`CountButtons component 1`] = `
"<div data-v-15663abe=\\"\\"><button id=\\"increment\\" type=\\"button\\" class=\\"actionBtn\\" data-v-15663abe=\\"\\">Minus(-)</button>
  <p data-v-15663abe=\\"\\">count is: 0</p><button id=\\"decrement\\" type=\\"button\\" class=\\"actionBtn\\" data-v-15663abe=\\"\\">Plus(+)</button>
</div>"
`;
